aiverify-foundation/moonshot
# [MS-339] Implemented fix for attack module endpoint table issue

## Description

Bug fix for attack modules. When an attack module uses an endpoint that is not created with a session (e.g. violent durian uses another gpt4 for toxic prompt generation), there will be an error. Refer to https://imda-dsl.atlassian.net/browse/MS-339 for more info

## Motivation and Context

Bug fix

## Type of Change
A bug fix

## How to Test

1. For moonshot main repo, pull from ms-339-am-endpoint-bug branch
2. For moonshot-data repo, pull from dev_main
3. Prepare two API keys: one endpoint and one gpt4 key and update the respective config files with the keys
4. Start CLI and create a session with any endpoint
    - `python -m moonshot cli interactive`
    - `new_session my-runner -e "['openai-gpt35-turbo']"`, or any other endpoint (except for gpt4)
5. Run attack module violent durian. You shouldn’t see the no such table error : 
    -`run_attack_module violent_durian "test"`

## Checklist

Please check all the boxes that apply to this pull request using "x":


## Screenshots (if applicable)

[If the changes involve visual modifications, include screenshots or GIFs that demonstrate the changes.]

## Additional Notes

[Add any additional information or context that might be relevant to reviewers.]

<details>
  <summary>Developer Certificate of Origin</summary>

 ```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
 ```
</details>
